openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -0,0 +1,100 @@
+"""
+Test audit user's access to various content based on content-gating features.
+"""
+
+import ddt
+from django.http import Http404
+from django.test.utils import override_settings
+from mock import Mock, patch
+
+from courseware.access import has_access
+from courseware.access_response import IncorrectPartitionGroupError
+from course_modes.tests.factories import CourseModeFactory
+from django.test.client import RequestFactory
+from lms.djangoapps.courseware.module_render import load_single_xblock
+from openedx.features.course_duration_limits.config import CONTENT_TYPE_GATING_FLAG
+from openedx.core.djangoapps.waffle_utils.testutils import override_waffle_flag
+from student.tests.factories import AdminFactory, CourseEnrollmentFactory, UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+
+@override_settings(
+    FIELD_OVERRIDE_PROVIDERS=(
+    'openedx.features.content_type_gating.field_override.ContentTypeGatingFieldOverride',)
+)
+@override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+@ddt.ddt
+class TestProblemTypeAccess(ModuleStoreTestCase):
+
+    def setUp(self):
```

Comment:
copying @calistenson 's code here

Reply:
Make sure that you rebase this code on my branch so that you're using the SharedModuleStoreTestCase inheritance and setUpClass method.